source/code/plugins/security_baseline_lib.rb

```diff
@@ -0,0 +1,116 @@
+                security_baseline_blob["DataItems"].push(oms_baseline_result)
+            end 
+
+            security_baseline_summary_blob = self.calculate_summary(results, host, time)
+
+            @@log.info "Security Baseline Summary: " + security_baseline_summary_blob.inspect
+
+            return security_baseline_blob, security_baseline_summary_blob
+        end # self.transform_and_wrap
+    
+        # ------------------------------------------------------   
+        def self.calculate_summary(results, hostname, time)  
+            asm_baseline_results = results["results"]
+            assessment_id = results["assessment_id"]
+            critical_failed_rules = 0;
+            warning_failed_rules = 0;
+            informational_failed_rules = 0;
+            all_failed_rules = 0;
+    
+            asm_baseline_results.each do |asm_baseline_result|
+
+                if asm_baseline_result["result"] == "PASS"  
+                    next 
+                end
+
+                all_failed_rules += 1
+
+                case asm_baseline_result["severity"]
+                when "Critical", "Important"
+                    critical_failed_rules += 1
+                when "Warning"
+                    warning_failed_rules += 1             
+                else
+                    informational_failed_rules += 1
+                end                   
+            end 
+
+            percentage_of_passed_rules = ((asm_baseline_results.length - all_failed_rules) * 100.0 / asm_baseline_results.length).round
+    
+            security_baseline_summary_blob = {
+                "DataType" => "SECURITY_BASELINE_SUMMARY_BLOB",
+                "IPName" => "Security",
+                "DataItems" => [
+                    {
+                        "Computer" => hostname,
+                        "TotalAssessedRules" => asm_baseline_results.length,
+                        "CriticalFailedRules" => critical_failed_rules,
+                        "WarningFailedRules" => warning_failed_rules,
+                        "InformationalFailedRules" => informational_failed_rules,
+                        "PercentageOfPassedRules" => percentage_of_passed_rules,
+                        "AssessmentId" => assessment_id,
+                        "OSName" => "Linux"
+                    }
+                ] 
+            }
+
+            return security_baseline_summary_blob
+        end # calculate_summary 
+
+        # ------------------------------------------------------
+        def self.transform_asm_2_oms(asm_baseline_result, scan_time, host, assessment_id)
+            oms = {
+                "TimeAnalyzed" => scan_time,
+                "Computer" => host,
+                "CceId"=> asm_baseline_result["cceid"],
+                "Severity" => asm_baseline_result["severity"],
+                "Name" => asm_baseline_result["description"],
+                "AnalyzeResult" => asm_baseline_result["result"] == "PASS" ? "Passed" : "Failed",
```

Comment:
distinguish between an error and failure required significant changes in omsbaseline and will be added in the next phase.